tawazi/dag.py

```diff
@@ -280,14 +142,15 @@ def _build(self) -> None:
                 f"the product contains at least a circular dependency: {cycle}"
             )
 
-        # 3. set sequence order
-        topological_order = self.graph_ids.topologically_sorted
-
-        # 4. calculate the sum of priorities of all recursive children
-        self._assign_compound_priority()
-
-        # 5. make a valid execution sequence to run sequentially if needed
-        self.exec_node_sequence = [self.node_dict[xn_id] for xn_id in topological_order]
+    def _validate(self) -> None:
+        # validate setup ExecNodes
+        for xn in self.exec_nodes:
+            if xn.setup:
```

Comment:
could we make a `and` rather than 2 nested ifs ?

Reply:
done